### ISSUE TITLE
Manually set mime types

### ISSUE THREAD

**Issue by geekygnr** [arangodb/arangodb#1702]
Hi,

I realize I am probably not using arangodb as intended but it makes sense for my use case.

I have a foxx application that serves an angularjs app which intern calls back to the foxx app itself. This application also needs to serve a lot of different types of files.

I have to do this because my entire application must be able to run on a standalone system, and I don't want to install a second web server just to serve the files.

I don't expect you to have a complete mime.types list for your web server (though I am surprised you don't just use the /etc/mime.types file). I just need to find a way to map some extensions to content types.

A couple of the extensions I tried and returned as text are *.JPG (but not *.jpg) and *.mov

Any help you could offer would be great.
Thanks

**Comment by dothebart** [comment]
You probably already saw our cookbook for handling binary data? https://docs.arangodb.com/cookbook/FoxxBinaryFileUploads.html - It demonstrates howto return binary data and set arbitrary headers (content-type)

Next to looking up the data in a pre-seeded mime database another good trick is to trust the content type header when uploading the file from a browser.

I will extend this example with an arangosh script to feed /etc/mime.types into a collection for lookup from FOXX in the next days - stay tuned.

**Comment by pluma** [comment]
You can use the `mime-types` module from npm (just npm-install it in your Foxx service folder as if it were a node app): https://www.npmjs.com/package/mime-types

Alternatively if you want to send files from Foxx you can also use `res.sendFile(filename)`, which will guess the content type the same way as files specified in the manifest.

**Comment by dothebart** [comment]
I've added https://github.com/arangodb/arangodb/blob/devel/scripts/read_mimetypes.js to read /etc/mime.types into a collection; However one problem persists: extension->mimetype is not uniq.

**Comment by pluma** [comment]
@dothebart Do we use that for anything at the moment? Also, do we use `arangodb.guessContentType` for anything other than Foxx at the moment? Because we're going to use `mime-types` from npm in most of FOTF to maintain consistency throughout npm modules, so if we only use it for Foxx we could phase it out entirely in the future (`mime-types` brings a full mime type list along).

**Comment by dothebart** [comment]
@pluma Its brand new. `arangodb.guessContentType` could probably also be used; However, if you want new mimetypes and don't want to alter the code base the mimetype collection could be the better way.

**Comment by geekygnr** [comment]
I am not using a foxx controller to deliver the files. I am just browsing to them directly. I have 

`"files": {
    "/html": {
      "path": "html/"
    }`

in my manifest.json and all the files I use are manually loaded into html/uploads (not entirely correct I know but I am dealing with a proof of concept and the client doesn't care how the files are entered).

I was more wondering if there was a way to configure the web server arangodb is using to detect additional mime types.

**Comment by pluma** [comment]
@geekygnr ArangoDB currently uses a hardcoded mime type list to guess the mime type based on extension when serving entire folders or not specifying the mime type:

https://github.com/arangodb/arangodb/blob/2.8/js/common/modules/org/arangodb/mimetypes.js#L14

FOTF (the upcoming major version of Foxx) will use `mime-db` (via `mime-types`), which pretty much ships the entire kitchen sink of mime type mappings (via Apache and nginx): https://github.com/jshttp/mime-db/tree/master/src This also includes the mime type for "mov".

Currently there is no way to make arangodb recognize additional mime types. I'm not sure how valuable tapping into this for the `files` manifest field is as this usage seems like an edge case. You probably don't want to serve file uploads from ArangoDB directly, at least not in production and automatic detection with a more complete mime type list should be good enough for your specific use case.

I noticed there is no way to specify the mime type of an individual file in the manifest, so it would probably be useful to be able to specify the type explicitly (via a `type` field which I thought already existed). I've added a task for that to the FOTF PR.
